docs(heroku): add note about necessary `package.json` scripts (#377)
unjs/nitro: docs/deploy/providers/heroku.md

   ```bash
   heroku config:set NITRO_PRESET=heroku
   ```

> 1. Add `start` and `heroku-postbuild` commands to `package.json` file.

Reviewer: ```suggestion

1. Add `start` and `heroku-postbuild` commands to `package.json` file.
```

   ```
   "scripts": {

>       "start": "node .output/server/index.mjs"

Reviewer: ```suggestion
     "start": "node .output/server/index.mjs"
```

   ```bash
   heroku config:set NITRO_PRESET=heroku
   ```

> 1. Add `start` and `heroku-postbuild` commands to `package.json` file.

Reviewer: ```suggestion
1. Ensure you have `start` and `build` commands in your `package.json` file.
```

> 1. Ensure you have `start` and `build` commands in your `package.json` file.

Reviewer 2: ```suggestion
1. Ensure you have `start` and `build` commands in your `package.json` file. (build command is either `nitro build` or `nuxt build`)
```

>    ```

Reviewer: ```suggestion
   ```json5
```

Reviewer 2: package.json does not allow trailing commas and comments. it can be nice for documentation purpose but also tricky for lazy copy pasters

   "scripts": {

>      "start": "node .output/server/index.mjs"

Reviewer: Tricky to make this framework agnostic - what do you think @pi0?

```suggestion
     "build": "nitro build", // or `nuxt build` if using nuxt
     "start": "node .output/server/index.mjs"
```

   ```bash
   heroku config:set NITRO_PRESET=heroku
   ```

>    
> 1. Configure your app.
> 
>    ```bash
>    heroku config:set NITRO_PRESET=heroku
>    ```
>    
> 
> ## Update your config
> 
> In order to run Nuxt 3 on Heroku, you need to update the following packages.
> 

Reviewer: Duplicated.
```suggestion
```

   ```bash
   heroku config:set NITRO_PRESET=heroku
   ```
   
1. Configure your app.

   ```bash
   heroku config:set NITRO_PRESET=heroku
   ```
   

> 1. Configure your app.
> 
>    ```bash
>    heroku config:set HOST=0.0.0.0
>    ```

Reviewer: This should not be necessary as `HOST` defaults to `0.0.0.0`

```suggestion
```

## Update your config

In order to run Nuxt 3 on Heroku, you need to update the following packages.

> 1. Add to `nuxt.config.ts` file the `nitro` preset.
> 
>    ```
>    import { defineNuxtConfig } from 'nuxt';
> 
>    // https://v3.nuxtjs.org/api/configuration/nuxt.config
>    export default defineNuxtConfig({
>      nitro: {
>        preset: 'node-server',
>      },
>    });
>    ```

Reviewer: ```suggestion
```

1. Add `start` and `heroku-postbuild` commands to `package.json` file.

   ```
   "scripts": {

>       "heroku-postbuild": "npm run build",

Reviewer: Is this required? Does heroku not run the `build` script automatically?

Reviewer 2: Docs: https://devcenter.heroku.com/changelog-items/1557

Reviewer: ```suggestion
```

>       "start": "nuxt preview"

Reviewer: ```suggestion
      "start": "node .output/server/index.mjs"
```

Reviewer 2: How was it working before without start @danielroe ?

Reviewer: Likely because the starter package had `"start": "node .output/server/index.mjs"` already present.

   }
   ```

> 1. Move the `nuxt` package from `devDependencies` to `dependencies` or Heroku won't be able to run command `npm start` because it runs under the `PRODUCTION` environment.
> 
>    ```
>    "dependencies": {
>       "nuxt": "3.0.0-rc.6"
>    }
>    ```

Reviewer: ```suggestion
```
